### ISSUE TITLE
[Feature Request] Chapter previews

### ISSUE THREAD

**Issue by H-yphen** [inorichi/tachiyomi#1632]
Hey, great work on the app. I can't imagine reading manga without it. 
One thing I miss though is the ability to zoom out and preview an entire chapter, with each page being a small thumbnail.
This allows for quickly jumping between pages in a visual way instead of using the slider and trying to remember the page number.

**Comment by metroid112** [comment]
Sounds like a nice feature, but trying to load all images at once from the chapter might be a bit performance heavy specially on manga with long chapters. And it probably needs to be downloaded as well

**Comment by H-yphen** [comment]
> but trying to load all images at once from the chapter might be a bit performance heavy

Well, considering that most services readily provide downscaled page thumbnails, I imagine that attempting to fetch them shouldn't be complicated API-wise or data-wise _(considering that thumbnails hardly go over 50kB)._
But even if not so, different approaches could be taken to ensure little to no performance loss; 
Things like only loading previews when the chapter is locally downloaded or cached, 
or like loading no more than `n` pages at a time _(or `n before` and `n after` the current page)_ along with a `load more` button or gesture.
As long as there's no fundamental issue with trying to implement this feature, a relatively easy solution most likely exists for any other non-fundamental issue.

**Comment by CarlosEsco** [comment]
@h-yphen can you point some sites the downscale the actual pages?

**Comment by hyphenrf** [comment]
@CarlosEsco it's been a while! I'm not very familiar with exact names of that kind of software. But I know it exists! I think (sorry if my guess is completely wrong but this has been a small quick search) images.weserv.nl 
I can't pretend  I'm aware how difficult or tasking is it to use their API or to use the catalogue's native API itself (since many sites do show thumbnailed versions of the chapter pages on their websites.)
But if it's at all approachable, it's a suggestion to consider.

**Comment by arkon** [comment]
Suggested in #1901: have pages thumbnails as slider

**Comment by joseph619** [comment]
I wonder if this could be requested for 1.x. To be honest, the seekbar gives very little control while scrolling through pages, especially for long chapters. A grid thumbnail preview of pages would offer much more flexibility. 

This wouldn't work for slow sources like Mangadex, but I could see it working for sites like Kakalot/Jar/Window etc which have fast loading.

**Comment by MatteCrystal** [comment]
If speed of the app is an issue (I don't think it should be for many sites considering I can view gallery in browser fast) But if it is an issue for such a feature than I would argue you should implement one or all of the following.

1.A drop down toggle that is initially closed that will show gallery view. (nothing will be loaded unless user clicks to view gallery)
2.A setting to allow users to decide if they want to show the gallery by default. (Ok with the potential wait will turn it on) 
3.Only show the first 10-20 pages at once, then show a load more button.

**Comment by ZerOri** [comment]
Yes,just like Perfectviewer,it`s a good feature.But 3 years have passed and it is still not completed.😥😥😥

![image](https://user-images.githubusercontent.com/46775615/147352433-2e784c9b-f79f-4c08-b3e7-fd1a81037a05.png)
